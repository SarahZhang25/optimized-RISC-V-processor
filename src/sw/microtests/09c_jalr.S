// This is an even more complex test that tests that your processor handles
// JALR (and LUI and ADDI) correctly.
.text
start:
    li x1, 0x1111
    li x2, 0x2222
    jalr x3,  0x123(x1) // A. jumps to 0x1234
    li x16, 1
    unimp

. = 0x1234
    jalr x4,   0x247(x1) // B. jumps to 0x1358
    jalr x10,  0x77f(x1) // H. jumps to 0x1890
    jalr x12, -0x672(x2) // J. jumps to 0x1bb0
    unimp

. = 0x1358
    jalr x5,   0x777(x1) // C. jumps to 0x1888
    jalr x9,   0x127(x1) // G. jumps to 0x1238
    jalr x14, -0x66e(x2) // L. jumps to 0x1bb4
    unimp

. = 0x1888
    jalr x6,  -0x676(x2) // D. jumps to 0x1bac
    jalr x8,   0x24b(x1) // F. jumps to 0x135c
    jalr x11,  0x12b(x1) // I. jumps to 0x123c
    unimp

. = 0x1bac
    jalr x7,   0x77b(x1) // E. jumps to 0x188c
    jalr x13,  0x24f(x1) // K. jumps to 0x1360
    jalr x15,    0x0(x3) // M. return to 0xc
    unimp
