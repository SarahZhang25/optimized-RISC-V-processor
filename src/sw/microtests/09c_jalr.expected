Dumping the state of the processor
pc = 0x00000018
x0 = 0x00000000
x1 = 0x00001111
x2 = 0x00002222
x3 = 0x00000014
x4 = 0x00001238
x5 = 0x0000135c
x6 = 0x0000188c
x7 = 0x00001bb0
x8 = 0x00001890
x9 = 0x00001360
x10 = 0x0000123c
x11 = 0x00001894
x12 = 0x00001240
x13 = 0x00001bb4
x14 = 0x00001364
x15 = 0x00001bb8
x16 = 0x00000001
x17 = 0x00000000
x18 = 0x00000000
x19 = 0x00000000
x20 = 0x00000000
x21 = 0x00000000
x22 = 0x00000000
x23 = 0x00000000
x24 = 0x00000000
x25 = 0x00000000
x26 = 0x00000000
x27 = 0x00000000
x28 = 0x00000000
x29 = 0x00000000
x30 = 0x00000000
x31 = 0x00000000
Quitting simulation.
